thunder/core/module.py

```diff
@@ -26,7 +27,13 @@ def __init__(self, model, compiled_model_call):
         self._forward_fn = compiled_model_call
 
         # overrides for parameters and buffers (see get_buffer/get_parameter)
-        self._overrides = {}
+        # we populate these here for performance reasons (sam as module cache),
+        # a single dict lookup is cheaper than traversin the module
+        # hierarchy, see https://github.com/Lightning-AI/lightning-thunder/issues/396#issuecomment-2113231498
+        self._overrides = {
+            k: v for k, v in itertools.chain(self._model.named_parameters(), self._model.named_buffers())
+        }
```

Comment:
Have you considered using `nn.Module.state_dict()`? The only difference I see from the current implementation is that here both persistent and non-persistent buffers are included, while state_dict includes only persistent buffers.
Why is LitGPT using non-persistent buffers for `cos` and `sin`? https://github.com/Lightning-AI/litgpt/blob/5dabf5f6be507e4da4d7be69813e6f31e799aefd/litgpt/model.py#L52-L53
```suggestion
        self._overrides = self._model.state_dict() | {k: v for k, v in self._model.named_buffers()}
```

Reply:
state_dict can actually contain all sorts of things (versions, ...), so I think tailoring to parameters and buffers seems the right thing.
I think we don't want to have cos and sin in the state dict (why should we...)